src/components/TableFilter.js

```diff
@@ -148,138 +143,128 @@ class TableFilter extends React.Component {
+              ))}
+            </Select>
+          </FormControl>
+        </div>
+      </GridListTile>
     );
   }
 
-  renderTextField(columns) {
+  renderTextField(column, index) {
     const { classes, filterList } = this.props;
 
     return (
-      <div className={classes.textFieldRoot}>
-        {columns.map((column, index) =>
-          column.filter ? (
-            <FormControl className={classes.textFieldFormControl} key={index}>
-              <TextField
-                label={column.name}
-                value={filterList[index].toString() || ''}
-                onChange={event => this.handleTextFieldChange(event, index)}
-              />
-            </FormControl>
-          ) : (
-            false
-          ),
-        )}
-      </div>
+      <GridListTile key={index} cols={1}>
+        <div className={classes.textFieldRoot}>
+          <FormControl className={classes.textFieldFormControl} key={index}>
+            <TextField
+              label={column.name}
+              value={filterList[index].toString() || ''}
+              onChange={event => this.handleTextFieldChange(event, index)}
+            />
+          </FormControl>
+        </div>
+      </GridListTile>
     );
   }
 
-  renderMultiselect(columns) {
+  renderMultiselect(column, index) {
     const { classes, filterData, filterList, options } = this.props;
 
     return (
-      <div className={classes.selectRoot}>
-        {columns.map((column, index) =>
-          column.filter ? (
-            <FormControl className={classes.selectFormControl} key={index}>
-              <InputLabel htmlFor={column.name}>{column.label}</InputLabel>
-              <Select
-                multiple
-                value={filterList[index] || []}
-                renderValue={selected => selected.join(', ')}
-                name={column.name}
-                onChange={event => this.handleMultiselectChange(index, event.target.value)}
-                input={<Input name={column.name} id={column.name} />}>
-                {filterData[index].map((filterColumn, filterIndex) => (
-                  <MenuItem value={filterColumn} key={filterIndex + 1}>
-                    <Checkbox
-                      checked={filterList[index].indexOf(filterColumn) >= 0 ? true : false}
-                      value={filterColumn.toString()}
-                      className={classes.checkboxIcon}
-                      classes={{
-                        root: classes.checkbox,
-                        checked: classes.checked,
-                      }}
-                    />
-                    <ListItemText primary={filterColumn} />
-                  </MenuItem>
-                ))}
-              </Select>
-            </FormControl>
-          ) : (
-            false
-          ),
-        )}
-      </div>
+      <GridListTile key={index} cols={1}>
+        <div className={classes.selectRoot}>
+          <FormControl className={classes.selectFormControl} key={index}>
```

Comment:
Also some of this `key={index}` are redundant and React complains about it 
```
Warning: Each child in a list should have a unique "key" prop.
```

Reply:
Fixed it.